src/perf_counters.cc

```diff
@@ -0,0 +1,116 @@
+#include "perf_counters.h"
+
+#if defined HAVE_LIBPFM
+#include <cstring>
+
+#include "perfmon/pfmlib.h"
+#include "perfmon/pfmlib_perf_event.h"
+#endif
+
+#ifdef BENCHMARK_OS_WINDOWS
+size_t read(int, void*, size_t) { return 0; }
+#endif
+namespace benchmark {
+namespace internal {
+
+#if defined HAVE_LIBPFM
+const bool PerfCounters::kSupported = true;
+
+bool PerfCounters::Initialize() { return pfm_initialize() == PFM_SUCCESS; }
+
+PerfCounters PerfCounters::Create(
+    const std::vector<std::string>& counter_names) {
+  if (counter_names.empty()) {
+    return NoCounters();
+  }
+  if (counter_names.size() > PerfCounterValues::kMaxCounters) {
+    GetErrorLogInstance()
+        << counter_names.size()
+        << " counters were requested. The minimum is 1, the maximum is "
+        << PerfCounterValues::kMaxCounters << "\n";
+    return NoCounters();
+  }
+  std::vector<int> counter_ids(counter_names.size());
+
+  const int mode = PFM_PLM3;  // user mode only
+  for (size_t i = 0; i < counter_names.size(); ++i) {
+    const bool is_first = i == 0;
+    struct perf_event_attr attr;
+    std::memset(&attr, 0, sizeof(perf_event_attr));
```

Comment:
You can just do `perf_event_attr attr{};` without `memset()`.

Reply:
oh - yes. thanks!